mozilla_version/gecko.py

```diff
@@ -157,6 +157,15 @@ def __attrs_post_init__(self):
             ), (
                 self.beta_number is not None and self.patch_number is not None,
                 'Beta number and patch number cannot be both defined',
+            ), (
+                self.is_major and self.is_stability,
+                'Version cannot be both a major and a stability one',
+            ), (
+                self.is_major and self.is_development,
+                'Version cannot be both a major and a development one',
+            ), (
+                self.is_stability and self.is_development,
+                'Version cannot be both a stability and a development one',
```

Comment:
Based on the current implementation of these 3 new properties, this errors should never happen. That said, they just helped me get the right implementation. Moreover, they will be useful any time someone changes or subclasses these 3 properties.